src/pages/scientificServices/pipelines/views/RunJob.tsx

```diff
@@ -1,65 +1,153 @@
-import { ButtonPrimary, Select, Spinner } from '@terra-ui-packages/components';
+import { ButtonPrimary, Icon, Link, Select, Spinner } from '@terra-ui-packages/components';
 import { isEmpty } from 'lodash';
 import React, { useEffect, useState } from 'react';
+import { ClipboardButton } from 'src/components/ClipboardButton';
 import FooterWrapper from 'src/components/FooterWrapper';
-import { TextArea, TextInput } from 'src/components/input';
 import { getPopupRoot } from 'src/components/popup-utils';
 import { Teaspoons } from 'src/libs/ajax/teaspoons/Teaspoons';
-import { Pipeline } from 'src/libs/ajax/teaspoons/teaspoons-models';
+import { Pipeline, PipelineInput } from 'src/libs/ajax/teaspoons/teaspoons-models';
+import * as Nav from 'src/libs/nav';
+import { notify } from 'src/libs/notifications';
 import { useCancellation } from 'src/libs/react-utils';
 import { pipelinesTopBar } from 'src/pages/scientificServices/pipelines/common/scientific-services-common';
+import { PipelineFileInput } from 'src/pages/scientificServices/pipelines/components/inputs/PipelineFileInput';
+import { PipelineRunDescription } from 'src/pages/scientificServices/pipelines/components/inputs/PipelineRunDescription';
+import { PipelineStringInput } from 'src/pages/scientificServices/pipelines/components/inputs/PipelineStringInput';
 import { HelpfulTipsWidget } from 'src/pages/scientificServices/pipelines/widgets/HelpfulTipsWidget';
 import { QuotaRemainingWidget } from 'src/pages/scientificServices/pipelines/widgets/QuotaRemainingWidget';
 
-async function uploadFileWithSignedUrl(inputFile, signedUrl) {
-  return await fetch(signedUrl, {
-    method: 'PUT',
-    headers: { 'Content-Type': 'application/octet-stream' },
-    body: inputFile,
+async function uploadFileWithSignedUrl(
```

Comment:
`fetch` is easier to work with, but unfortunately it doesn't provide access to some of the lower-level events that we need to listen for in order to show a progress bar, so i had to switch this over to `XMLHttpRequest`